New proton transport without data dir

mundim opened the pull request:
As agreed, following up the PR 24490 (which as closed), a new PR with the same packages but removing the data directory from SimTransport module.

mundim committed:
Splitting into PPSTools + new transport with totem param. and fastsim adaptation

mundim committed:
implementing PR suggestions

mundim committed:
implementing PR suggestions

mundim committed:
removing commented code that was not caught by PR testing

mundim committed:
removing typo

mundim committed:
Moving PPSTools/Utilities to Utilities/PPS as advised and the related changes in the code that depends on it

mundim committed:
followig up PR comments

mundim committed:
followig up PR comments

mundim committed:
Changing name and moving some CLHEP include to .cc

mundim committed:
Removing CLHEP includes from header files

mundim committed:
removing self assignment in HectorTransport.cc

mundim committed:
Applying requested corrections

mundim committed:
Removing data dir

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25355/7386

cmsbuild commented:
A new Pull Request was created by  @mundim  for master.

It involves the following packages:

FastSimulation/CTPPSFastTrackingProducer
FastSimulation/CTPPSRecHitProducer
FastSimulation/CTPPSSimHitProducer
IOMC/ParticleGuns
SimPPS/PPSSimTrackProducer
SimTransport/HectorProducer
SimTransport/PPSProtonTransport
SimTransport/TotemRPProtonTransportParametrization
Utilities/PPS


The following packages do not have a category, yet:

Utilities/PPS
Please create a PR for https://github.com/cms-sw/cms-bot/blob/master/categories_map.py to assign category

@ssekmen, @lveldere, @perrozzi, @efeyazgan, @mdhildreth, @cmsbuild, @civanch, @alberto-sanchez, @qliphy can you please review it and eventually sign? Thanks.
@Martin-Grunewald, @matt-komm, @wddgit this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

fabiocos commented:
please test with cms-sw/cmsdist#4515

cmsbuild commented:
The tests are being triggered in jenkins.
Using externals from cms-sw/cmsdist#4515

cmsbuild commented:
+1
Tested at: 0f0af28c9c3875ec442b6341453d50cf2726801c
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25355/31893/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25355/31893/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 3 differences found in the comparisons
 - DQMHistoTests: Total files compared: 33
 - DQMHistoTests: Total histograms compared: 3131839
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3131634
 - DQMHistoTests: Total skipped: 204
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 32 files compared)
 - Checked 137 log files, 14 edm output root files, 33 DQM output files

civanch commented:
+1
proposed modification for #24490 are implemented.

civanch commented:
@efeyazgan , this is long standing PR, whihc seems to converge now. Can you, please, sign it again?

efeyazgan commented:
+1

fabiocos commented on IOMC/ParticleGuns/src/RandomtXiGunProducer.cc:
```diff
@@ -35,6 +35,7 @@ RandomtXiGunProducer::RandomtXiGunProducer(const edm::ParameterSet& pset) :
    fMaxt = pgun_params.getParameter<double>("Maxt");
    fMinXi= pgun_params.getParameter<double>("MinXi");
    fMaxXi= pgun_params.getParameter<double>("MaxXi");
+   fLog_t= pgun_params.getUntrackedParameter<bool>("Log_t",false);
```
@mundim @civanch as this flag determines which kind of distribution is really simulated, would not be better to have it as a tracked parameter?

mundim replied:
By default, it assumes a linear distribution. The untracked parameter was used to keep backward compatibility with other config which do not use this parameter.

fabiocos commented on SimTransport/PPSProtonTransport/src/HectorTransport.cc:
```diff
@@ -0,0 +1,238 @@
+        return false;
+    }
+    if (m_verbosity) {
+          std::cout  << "====================================================================\n"
```
@mundim @civanch cout is not thread safe, could you please replace with LogInfo or LogVerbatim for instance?

fabiocos commented on SimTransport/TotemRPProtonTransportParametrization/src/LHCOpticsApproximator.cc:
```diff
@@ -0,0 +1,1212 @@
+    }
+  }
+
+  std::cout<<"Optical functions parametrizations from "<<s_begin_<<" to "<<s_end_<<std::endl;
```
@mundim @civanch also here could we replace std::cout with the MessageLogger?

mundim replied:
This code comes from TOTEM and we would like to keep it as close to the original as possible. New developments will be done in such a way it works both in the TOTEM alone and CMS framework, so, we would strongly like to avoid including anything from CMS since the more we change it, the more difficult it will be to include future developments.
So, would it be too bad to keep std::cout instead of using MessageLogger? Thanks

mundim replied:
the other comment is implemented. I'm just waiting for the reply on the request above to push the new code to avoid restarting all the change and then have to push again.

mundim replied:
Ok. I'll proceed with the needed changes. Thanks.

fabiocos commented:
@mundim apart for my small technical comments on the code, do you have some documentation of it to link in the description of the PR? In which workflows is this code supposed to run? You provide customisation files that looks and addition to regular workflows, do you have a test where do they run and this code is effectively probed?

mundim commented:
> @mundim, I agree with Fabio technical comments. They should be fixed in this or in the next PR.
> 
> @fabiocos, Concerning general situation with PPS simulation I would go step by step. I do not think, that it is final PR. The history is started in summer and we stack for a long time. Probably, even these technical fixes may be done in the following PRs. For example, it is strongly needed, that PPS geometry belong to a separate G4Region and is not part of World volume, because in CMS simulation all special features like time cuts are defined per G4Region. Making unit test or special WF should be a part of new PRs.

Yes, we have agreed that a dedicated G4Region should be defined in the next PR and I'm already wondering if this code will ever be approved. I'll try to implement Fabio's comments today. Thanks for your help.

mundim committed:
replacing std::cout by LogInfo in HectorTransport.cc

fabiocos commented:
@mundim @civanch ok let's integrate this PR as intermediate step, but I expect that my comments are addressed in the next one.

fabiocos commented:
+1

fabiocos commented:
@smuzaffar @civanch I think that the pending issue is the assignment of Utilities/PPS (all signatures are there, and the needed external has been merged already). That package has nothing to do with core, to me it looks like a simulation package and should be assigned to that category, do you agree?

davidlange6 commented:
hi @fabiocos

looking again at this pr I see little has changed in the random extra includes in header files - thats unfortunate. 

otherwise, do we really want something like this

void LorentzBoost(CLHEP::HepLorentzVector& p_out, const std::string& frame);

(both using obsolete chep vectors and strings)

> On Dec 4, 2018, at 9:41 AM, Fabio Cossutti <notifications@github.com> wrote:
> 
> @smuzaffar @civanch I think that the pending issue is the assignment of Utilities/PPS (all signatures are there, and the needed external has been merged already). That package has nothing to do with core, to me it looks like a simulation package and should be assigned to that category, do you agree?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

civanch commented:
@fabiocos , we excluded core from Utilities/PPS

mundim commented:
> @mundim @civanch ok let's integrate this PR as intermediate step, but I expect that my comments are addressed in the next one.

@fabiocos, please, wait a few hours and I'll push the code with your changes. thanks

fabiocos commented:
hold

cmsbuild commented:
Pull request has been put on hold by  @fabiocos
They need to issue an `unhold` command to remove the `hold` state or L1 can `unhold` it for all

fabiocos commented:
-1

waiting for a further update by @mundim here instead of the next PR

@civanch Utilities/PPS was not assigned for what I could see, ok not assigning it to core, but we need to assign it somewhere else, and simulation looks the most appropriate way to me

@davidlange6 thanks for your comments, indeed this needs to be addressed, I understood that @civanch was to move by steps. Let me come back to this

mundim committed:
Removing output to cout and replacing CLHEP HepLorentzVector by the ROOT version

cmsbuild commented:
The code-checks are being triggered in jenkins.

mundim commented:
Comments from @fabiocos and @davidlange6 implemented. Some changes about CLHEP could not (yet) be implemented because the dependency tree spread over other packages.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25355/7467

cmsbuild commented:
Pull request #25355 was updated. @ssekmen, @lveldere, @perrozzi, @efeyazgan, @mdhildreth, @cmsbuild, @civanch, @alberto-sanchez, @qliphy can you please check and sign again.

civanch commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: f2efc82bede7bc630759dd096366e70db8860494
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25355/32043/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25355/32043/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 33
 - DQMHistoTests: Total histograms compared: 3136422
 - DQMHistoTests: Total failures: 2
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3136216
 - DQMHistoTests: Total skipped: 204
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 32 files compared)
 - Checked 137 log files, 14 edm output root files, 33 DQM output files

civanch commented:
+1

civanch commented:
@efeyazgan , can you, please, sign this PR again - some minor technical improvements were added.

efeyazgan commented:
+1

civanch commented:
@fabiocos , it seems , that all requests are fulfilled.

fabiocos commented on Utilities/PPS/interface/PPSOpticsCalibrator.h:
```diff
@@ -0,0 +1,86 @@
+#include "TDirectory.h"
+#include "TSystem.h"
+#include "TRandom3.h"
+#include "TFile.h"
```
@mundim @civanch it looks like that the comment of @davidlange6 in #24490 has been ignored here, why?

fabiocos commented:
@mundim thank you for the update, I would like to get rid of this PR asap, I have a few remarks following the comments of @civanch and @davidlange6 , looking at the various issues mentioned:

- CLHEP vectors have been removed but from TotemTransport, why?

- CLHEP is used for units, but this is still true in many packages scattered around CMSSW, so I think that a general solution should be found. 

- I still see plenty of "cout": in what appears as a standalone utility as Utilities/PPS/bin/PPSOpticsCalibrator.cc it is not a problem, but what about SimTransport/TotemRPProtonTransportParametrization/src/TMultiDimFet.cc ? How is this code supposed to be used? Is it supposed to enter into a multi-threaded workflow? It looks like that Totem related code could be used in SimTransport/PPSProtonTransport/python/TotemTransport_cfi.py am I correct?

- the comment of @davidlange6 about the use of TFile.h seems to have been overlooked (or ignored, up to the author clarify)

mundim committed:
removing cout/CLHEP/TFile that were somehow left behind

mundim committed:
removing remaining CLHEP Vector

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25355/7753

- This PR adds an extra 1052KB to repository

- Found files with invalid states:
  - SimTransport/TotemRPProtonTransportParametrization/data/parametrization_6500GeV_0p4_185_reco_beam1.root: Added, Deleted
  - SimTransport/PPSProtonTransport/data/LHCB1_Beta0.40_6.5TeV_CR191.541_PreTS2.tfs: Added, Deleted
  - PPSTools/Utilities/interface/PPSOpticsCalibrator.h: Added, Deleted
  - Utilities/PPS/bin/BuildFile.xml: Added, Modified, Deleted
  - SimTransport/PPSProtonTransport/data/LHCB2_Beta0.30_6.5TeV_CR175_Nominal_2017.tfs: Added, Deleted
  - SimTransport/PPSProtonTransport/data/LHCB2_Beta0.40_6.5TeV_CR185_Nominal_2016.tfs: Added, Deleted
  - SimTransport/PPSProtonTransport/data/LHCB1_Beta0.40_6.5TeV_CR150_Nominal_2017.tfs: Added, Deleted
  - Utilities/PPS/bin/PPSOpticsCalibration.cc: Added, Deleted
  - SimTransport/TotemRPProtonTransportParametrization/data/parametrization_6500GeV_90_reco.root: Added, Deleted
  - SimTransport/PPSProtonTransport/data/LHCB2_Beta0.40_6.5TeV_CR179.394_PreTS2_TOTEM.tfs: Added, Deleted
  - SimTransport/PPSProtonTransport/data/LHCB1_Beta0.40_6.5TeV_CR191.541_PreTS2_TOTEM.tfs: Added, Deleted
  - SimTransport/PPSProtonTransport/data/LHCB2_Beta0.40_6.5TeV_CR179.394_PreTS2.tfs: Added, Deleted
  - SimTransport/TotemRPProtonTransportParametrization/data/parametrization_6500GeV_90_transp.root: Added, Deleted
  - PPSTools/Utilities/src/PPSUtilities.cc: Added, Deleted
  - PPSTools/Utilities/bin/BuildFile.xml: Added, Deleted
  - SimTransport/PPSProtonTransport/data/LHCB1_Beta0.40_6.5TeV_CR185_Nominal_2016.tfs: Added, Deleted
  - Utilities/PPS/interface/PPSOpticsCalibrator.h: Added, Modified, Deleted
  - SimTransport/PPSProtonTransport/data/LHCB2_Beta0.40_6.5TeV_CR150_Nominal_2017.tfs: Added, Deleted
  - PPSTools/Utilities/bin/PPSOpticsCalibration.cc: Added, Deleted
  - PPSTools/Utilities/interface/PPSUtilities.h: Added, Deleted
  - SimTransport/TotemRPProtonTransportParametrization/data/parametrization_6500GeV_90_transp_75.root: Added, Deleted
  - SimTransport/PPSProtonTransport/data/LHCB1_Beta0.30_6.5TeV_CR175_Nominal_2017.tfs: Added, Deleted
  - SimTransport/TotemRPProtonTransportParametrization/data/parametrization_6500GeV_90p0_50urad_reco.root: Added, Deleted
  - Utilities/PPS/bin/PPSOpticsCalibrator.cc: Added, Modified, Deleted
  - PPSTools/Utilities/BuildFile.xml: Added, Deleted
  - SimTransport/TotemRPProtonTransportParametrization/data/parametrization_6500GeV_0p4_185_reco_beam2.root: Added, Deleted
  - PPSTools/Utilities/interface/PPSUnitConversion.h: Added, Modified, Deleted

cmsbuild commented:
Pull request #25355 was updated. @efeyazgan, @lveldere, @perrozzi, @ssekmen, @mdhildreth, @cmsbuild, @civanch, @alberto-sanchez, @qliphy can you please check and sign again.

civanch commented:
@smuzaffar , can you, please, comment on the recent update of this PR: how to avoid addition of extra 1052KB to repository?

civanch commented:
@mundim, there is the problem with this PR as you see above. You need clean up the history or create a new PR for 10_5_X where manually copy all necessary files.

mundim commented:
> @civanch , as mentioned in the [#25355 (comment)](https://github.com/cms-sw/cmssw/pull/25355#issuecomment-449123560) , there are many files which were added/removed. This also contains some data files. Please get the history of this Pr cleaned up and force push the changes. Or create a new PR with clean history.

I don't understand what is going on. I have not added any file, just removed the ones in the bin directory which was just causing problem and it is not important they go into the release. See below:

git status
On branch newProtonTransport_NoDataDir
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

	modified:   FastSimulation/CTPPSFastTrackingProducer/plugins/CTPPSFastTrackingProducer.cc
	modified:   FastSimulation/CTPPSRecHitProducer/plugins/CTPPSRecHitProducer.cc
	modified:   IOMC/ParticleGuns/src/RandomtXiGunProducer.cc
	modified:   SimTransport/HectorProducer/src/Hector.cc
	modified:   SimTransport/HectorProducer/src/HectorProducer.cc
	modified:   SimTransport/PPSProtonTransport/interface/HectorTransport.h
	modified:   SimTransport/PPSProtonTransport/src/HectorTransport.cc
	modified:   SimTransport/PPSProtonTransport/src/ProtonTransport.cc
	modified:   SimTransport/PPSProtonTransport/src/TotemTransport.cc
	modified:   SimTransport/TotemRPProtonTransportParametrization/interface/LHCOpticsApproximator.h
	modified:   SimTransport/TotemRPProtonTransportParametrization/src/LHCOpticsApproximator.cc
	modified:   SimTransport/TotemRPProtonTransportParametrization/src/TMultiDimFet.cc
	deleted:    Utilities/PPS/bin/BuildFile.xml
	deleted:    Utilities/PPS/bin/PPSOpticsCalibrator.cc
	deleted:    Utilities/PPS/interface/PPSOpticsCalibrator.h
	modified:   Utilities/PPS/src/PPSUtilities.cc

fabiocos commented:
@smuzaffar this looks another example of the reports to be understood about added/removed file sizes. If I merge this PR as it is on top of CMSSW_10_5_X and look at the added/removed file sizes I see:

30      FastSimulation/CTPPSFastTrackingProducer/BuildFile.xml
-359    FastSimulation/CTPPSFastTrackingProducer/interface/CTPPSFastTrackingProducer.h
30      FastSimulation/CTPPSFastTrackingProducer/plugins/BuildFile.xml
-1205   FastSimulation/CTPPSFastTrackingProducer/plugins/CTPPSFastTrackingProducer.cc
22      FastSimulation/CTPPSFastTrackingProducer/python/CTPPSFastTrackingProducer_cfi.py
-18     FastSimulation/CTPPSRecHitProducer/plugins/CTPPSRecHitProducer.cc
160     FastSimulation/CTPPSRecHitProducer/python/CTPPSRecHitProducer_cfi.py
161     FastSimulation/CTPPSSimHitProducer/plugins/CTPPSSimHitProducer.cc
10      FastSimulation/CTPPSSimHitProducer/python/CTPPSSimHitProducer_cfi.py
30      IOMC/ParticleGuns/interface/BaseRandomtXiGunProducer.h
332     IOMC/ParticleGuns/src/RandomtXiGunProducer.cc
313     SimPPS/PPSSimTrackProducer/plugins/BuildFile.xml
7128    SimPPS/PPSSimTrackProducer/plugins/PPSSimTrackProducer.cc
2169    SimPPS/PPSSimTrackProducer/python/SimTrackProducerForFastSim_cff.py
1316    SimPPS/PPSSimTrackProducer/python/SimTrackProducerForFullSim_cff.py
-265867 SimTransport/HectorProducer/data/LHCB1IR5_5TeV.tfs
-265866 SimTransport/HectorProducer/data/LHCB1IR5_7TeV.tfs
-265866 SimTransport/HectorProducer/data/LHCB1IR5_v6.500.tfs
-265867 SimTransport/HectorProducer/data/LHCB2IR5_5TeV.tfs
-265866 SimTransport/HectorProducer/data/LHCB2IR5_7TeV.tfs
-265866 SimTransport/HectorProducer/data/LHCB2IR5_v6.500.tfs
-4143   SimTransport/HectorProducer/interface/CTPPSHector.h
-716    SimTransport/HectorProducer/interface/CTPPSHectorParameters.h
-1437   SimTransport/HectorProducer/interface/CTPPSHectorProducer.h
-2046   SimTransport/HectorProducer/python/FastSimWithCTPPS_cff.py
-1235   SimTransport/HectorProducer/python/FullSimWithCTPPS_cff.py
-1587   SimTransport/HectorProducer/python/HectorTransportCTPPS_cfi.py
-19770  SimTransport/HectorProducer/src/CTPPSHector.cc
-4269   SimTransport/HectorProducer/src/CTPPSHectorProducer.cc
-12     SimTransport/HectorProducer/src/Hector.cc
51      SimTransport/HectorProducer/src/HectorProducer.cc
419     SimTransport/PPSProtonTransport/BuildFile.xml
2561    SimTransport/PPSProtonTransport/interface/HectorTransport.h
1953    SimTransport/PPSProtonTransport/interface/ProtonTransport.h
2465    SimTransport/PPSProtonTransport/interface/TotemTransport.h
5147    SimTransport/PPSProtonTransport/python/HectorOpticsParameters_cfi.py
823     SimTransport/PPSProtonTransport/python/HectorTransport_cfi.py
2155    SimTransport/PPSProtonTransport/python/TotemBeamConditions_cff.py
460     SimTransport/PPSProtonTransport/python/TotemTransport_cfi.py
11117   SimTransport/PPSProtonTransport/src/HectorTransport.cc
4155    SimTransport/PPSProtonTransport/src/ProtonTransport.cc
9092    SimTransport/PPSProtonTransport/src/TotemTransport.cc
147     SimTransport/TotemRPProtonTransportParametrization/BuildFile.xml
8218    SimTransport/TotemRPProtonTransportParametrization/interface/LHCOpticsApproximator.h
11460   SimTransport/TotemRPProtonTransportParametrization/interface/TMultiDimFet.h
41050   SimTransport/TotemRPProtonTransportParametrization/src/LHCOpticsApproximator.cc
568     SimTransport/TotemRPProtonTransportParametrization/src/SimTransportTotemRPProtTranspParLinkDef.h
73263   SimTransport/TotemRPProtonTransportParametrization/src/TMultiDimFet.cc
187     Utilities/PPS/BuildFile.xml
807     Utilities/PPS/interface/PPSUnitConversion.h
964     Utilities/PPS/interface/PPSUtilities.h
2952    Utilities/PPS/src/PPSUtilities.cc
total -1440280

so apparently there is a reduction in the size of the master after this commit of about 1 MB. In my understanding what effectively matters from the point of view of the repository size is the fresh new additions, as the files removed are anyway present in the history. This should be 191715 bytes

fabiocos commented:
@smuzaffar @mundim ok, the report I show is for the final added snapshot, but the intermediate history that can be looked at with a cms-checkout-topic shows that this PR is built on top of CMSSW_10_3_0_pre3, and that the addition of the unwanted files is brought inside in the initial commits:

15:03 cmsdev15 784> git log -30 --pretty=oneline
53c31a74083f93a2713240625d0959a12eded8bd (HEAD -> pull/25355, cms-sw/refs/pull/25355/head) removing remaining CLHEP Vector
6f11bf02d0b7b1214d68f814dd6b2a0e1d6f13a1 removing cout/CLHEP/TFile that were somehow left behind
f2efc82bede7bc630759dd096366e70db8860494 Removing output to cout and replacing CLHEP HepLorentzVector by the ROOT version
dba9d40a499b696d6a702e8c73f9bb97c3bf4ced replacing std::cout by LogInfo in HectorTransport.cc
0f0af28c9c3875ec442b6341453d50cf2726801c Removing data dir
2be631b5dc7bf2229bff8c9b297fb967eb85bba7 Applying requested corrections
1b1d60b387d06483781e2dc53c0a754534201674 removing self assignment in HectorTransport.cc
7120ddbe61c3191052fa55416e302efc38177bfc Removing CLHEP includes from header files
f3d483e1d5074db78a5ba70ee8077d1bd872647c Changing name and moving some CLHEP include to .cc
971c9548b3ee6b5b7db8061442bd8d02683637ff followig up PR comments
3e60d925ce9f57e1455c79677a34525943cbdc27 followig up PR comments
4c192080696da6c6ce0f8f3c325aa85a7b80f0e2 Moving PPSTools/Utilities to Utilities/PPS as advised and the related changes in the code that depends on it
79bac8e0b8d7d4a34f7d41628ca75dcc79ab42f8 removing typo
ff43c22fd62d9f9b1f1e1f5a5b5802bed20c74be removing commented code that was not caught by PR testing
99bcc27e5bb4bcca4f81747c3184e3391016f7ab implementing PR suggestions
97e2ad74ecf69d50f1393d2c60f33e589313ba81 implementing PR suggestions
43e133b1de8235061a094dff89b062db0f99d1c2 Splitting into PPSTools + new transport with totem param. and fastsim adaptation
79bbbf12484188ad2c724f3a27a19759efe5dcbf (tag: CMSSW_10_3_0_pre3) Merge pull request #24468 from slava77/CMSSW_10_3_X_2018-09-04-1100/sign1038/dpHI2018

Between 79bbbf12484188ad2c724f3a27a19759efe5dcbf and 43e133b1de8235061a094dff89b062db0f99d1c2 8237049 bytes are added, including the root files, while between 2be631b5dc7bf2229bff8c9b297fb967eb85bba7 and 0f0af28c9c3875ec442b6341453d50cf2726801c 9640757 bytes are removed (all the data files moved into the external package). 

In order not to lose the history of the review i would suggest to open a new PR where the content of this one is squashed (so as the intermediate history is cleaned) and rebased on top of 10_5_X.

fabiocos commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 53c31a74083f93a2713240625d0959a12eded8bd
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25355/32435/summary.html

smuzaffar commented:
code-check

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25355/7845

- This PR adds an extra 1804KB to repository

- Found files with invalid states:
  - SimTransport/TotemRPProtonTransportParametrization/data/parametrization_6500GeV_0p4_185_reco_beam1.root: Added, Deleted
  - SimTransport/PPSProtonTransport/data/LHCB1_Beta0.40_6.5TeV_CR191.541_PreTS2.tfs: Added, Deleted
  - PPSTools/Utilities/interface/PPSOpticsCalibrator.h: Added, Deleted
  - Utilities/PPS/bin/BuildFile.xml: Added, Modified, Deleted
  - SimTransport/PPSProtonTransport/data/LHCB2_Beta0.30_6.5TeV_CR175_Nominal_2017.tfs: Added, Deleted
  - SimTransport/PPSProtonTransport/data/LHCB2_Beta0.40_6.5TeV_CR185_Nominal_2016.tfs: Added, Deleted
  - SimTransport/PPSProtonTransport/data/LHCB1_Beta0.40_6.5TeV_CR150_Nominal_2017.tfs: Added, Deleted
  - Utilities/PPS/bin/PPSOpticsCalibration.cc: Added, Deleted
  - SimTransport/TotemRPProtonTransportParametrization/data/parametrization_6500GeV_90_reco.root: Added, Deleted
  - SimTransport/PPSProtonTransport/data/LHCB2_Beta0.40_6.5TeV_CR179.394_PreTS2_TOTEM.tfs: Added, Deleted
  - SimTransport/PPSProtonTransport/data/LHCB1_Beta0.40_6.5TeV_CR191.541_PreTS2_TOTEM.tfs: Added, Deleted
  - SimTransport/PPSProtonTransport/data/LHCB2_Beta0.40_6.5TeV_CR179.394_PreTS2.tfs: Added, Deleted
  - SimTransport/TotemRPProtonTransportParametrization/data/parametrization_6500GeV_90_transp.root: Added, Deleted
  - PPSTools/Utilities/src/PPSUtilities.cc: Added, Deleted
  - PPSTools/Utilities/bin/BuildFile.xml: Added, Deleted
  - SimTransport/PPSProtonTransport/data/LHCB1_Beta0.40_6.5TeV_CR185_Nominal_2016.tfs: Added, Deleted
  - Utilities/PPS/interface/PPSOpticsCalibrator.h: Added, Modified, Deleted
  - SimTransport/PPSProtonTransport/data/LHCB2_Beta0.40_6.5TeV_CR150_Nominal_2017.tfs: Added, Deleted
  - PPSTools/Utilities/bin/PPSOpticsCalibration.cc: Added, Deleted
  - PPSTools/Utilities/interface/PPSUtilities.h: Added, Deleted
  - SimTransport/TotemRPProtonTransportParametrization/data/parametrization_6500GeV_90_transp_75.root: Added, Deleted
  - SimTransport/PPSProtonTransport/data/LHCB1_Beta0.30_6.5TeV_CR175_Nominal_2017.tfs: Added, Deleted
  - SimTransport/TotemRPProtonTransportParametrization/data/parametrization_6500GeV_90p0_50urad_reco.root: Added, Deleted
  - Utilities/PPS/bin/PPSOpticsCalibrator.cc: Added, Modified, Deleted
  - PPSTools/Utilities/BuildFile.xml: Added, Deleted
  - SimTransport/TotemRPProtonTransportParametrization/data/parametrization_6500GeV_0p4_185_reco_beam2.root: Added, Deleted
  - PPSTools/Utilities/interface/PPSUnitConversion.h: Added, Modified, Deleted

smuzaffar commented:
@mundim , in this PR https://github.com/cms-sw/cmssw/pull/25355/commits/43e133b1de8235061a094dff89b062db0f99d1c2 commit adds some files while https://github.com/cms-sw/cmssw/pull/25355/commits/0f0af28c9c3875ec442b6341453d50cf2726801c remove those files. As these commits are part of this PR history so this is going to pollute cmssw repository. 

I agree with @fabiocos , please create a new PR with clean history.

smuzaffar commented:
hold

cmsbuild commented:
Pull request has been put on hold by  @smuzaffar, @fabiocos
They need to issue an `unhold` command to remove the `hold` state or L1 can `unhold` it for all

mundim commented:
As adivised, closing this PR and issuing a new one from a clean area, rebased to CMSSW_10_5
